cmd/restic/cmd_forget.go

```diff
@@ -108,10 +108,16 @@ func runForget(opts ForgetOptions, gopts GlobalOptions, args []string) error {
 		return err
 	}
 
-	lock, err := lockRepoExclusive(gopts.ctx, repo)
-	defer unlockRepo(lock)
-	if err != nil {
-		return err
+	if gopts.NoLock && !opts.DryRun {
+		Exitf(1, "Error: --no-lock is only applicable in combination with --dry-run for forget command")
```

Comment:
Please replace this with
```
return errors.Fatal("--no-lock is only applicable in combination with --dry-run for forget command")
```

I have to admit that there are some uses of `Exitf` in two other commands, however, returning a fatal error, as it's done in most places, is much cleaner in my opinion.